src/nextjournal/clerk/analyzer.clj

```diff
@@ -234,39 +253,41 @@
   (let [rec-form (concat '(do) [form (get-in state [:->analysis-info dependency :form])])
         rec-var (symbol (str/join "+" (sort (conj vars dep))))
         var (first vars)] ;; TODO: reduce below
+    ;;(when (not= dep dependency) (println :dep-mismatch dep dependency))
+    ;;(when (not= var node) (println :node-mismatch var node))
     (-> state
         (update :graph #(-> %
                             (dep/remove-edge dependency node)
                             (dep/depend var rec-var)
                             (dep/depend dep rec-var)))
         (assoc-in [:->analysis-info rec-var :form] rec-form))))
 
-
-(defn ->key [{:as codeblock :keys [var id]}]
-  (if var var id))
-
-(defn ->ana-keys [{:as analyzed :keys [form vars id]}]
-  (if (seq vars) vars [(->key analyzed)]))
-
 #_(->> (nextjournal.clerk.eval/eval-string "(rand-int 100) (rand-int 100) (rand-int 100)") :blocks (mapv #(-> % :result :nextjournal/value)))
 
-(defn- analyze-deps [{:as analyzed :keys [form vars]} state dep]
-  (try (reduce (fn [state _var] ;; TODO: check if `_var` needs to be used
-                 (update state :graph #(dep/depend % (->key analyzed) dep)))
-               state
-               (->ana-keys analyzed))
+(defn dep->block-id
+  "Inverse key-lookup from deps to block ids"
+  [{:keys [var->block-id]} dep]
+  (get var->block-id dep dep))
+
+(defn- analyze-deps [{:as _info :keys [id form vars]} state dep]
+  (try (update state :graph dep/depend id (dep->block-id state dep))
        (catch Exception e
          (if (circular-dependency-error? e)
            (analyze-circular-dependency state vars form dep (ex-data e))
            (throw e)))))
 
-
-(defn make-deps-inherit-no-cache [state {:as analyzed :keys [no-cache? vars deps ns-effect?]}]
-  (if no-cache?
-    state
-    (let [no-cache-deps? (boolean (some (fn [dep] (get-in state [:->analysis-info dep :no-cache?])) deps))]
-      (reduce (fn [state k]
-                (assoc-in state [:->analysis-info k :no-cache?] no-cache-deps?)) state (->ana-keys analyzed)))))
+(defn make-deps-inherit-no-cache [{:as doc :keys [blocks graph ->analysis-info]}]
```

Comment:
this is now called once, after the whole graph has been built, to also consider edges across files (#633).